src/packet/secret_key_macro.rs

```diff
@@ -150,6 +150,75 @@ macro_rules! impl_secret_key {
                     .build()?
                     .sign_key(key, key_pw, &self)
             }
+
+            /// Unlock the private key material of this secret key packet.
+            /// This removes password protection ("locking") of the secret key material.
+            ///
+            /// If the Secret Key packet is not locked, it is left unchanged.
+            ///
+            /// The current locking password for this key must be provided in `password`.
+            pub fn password_unlock<P>(&mut self, password: P) -> $crate::errors::Result<()>
```

Comment:
bikeshedding: `unlock` to me sounds very temporary

Reply:
I see what you mean. The RFC uses "lock" and "unlock", and I like that use of these two terms is specific to s2k-based protection of secret key material.

However, the RFC only really uses "unlock" in the context of a temporary operation. So I guess you're right.
Names like `remove_password` and `set_password` are probably better, after all.

Reply:
pushed a naming change (and more explicit documentation for the "change password" use case)